tests/src/python/test_qgsserver.py

```diff
@@ -166,6 +168,79 @@ def test_project_wms(self):
+
+        query_string = 'MAP=%s&SERVICE=WFS&VERSION=1.1.0&REQUEST=%s' % (urllib.quote(project), request)
+        header, body = [str(_v) for _v in self.server.handleRequest(query_string)]
+        self.assert_headers(header, body)
+        response = header + body
+        f = open(self.testdata_path + 'wfs_'+ request.lower() + '.txt')
+        expected = f.read()
+        f.close()
+        # Store the output for debug or to regenerate the reference documents:
+        """
+        f = open(os.path.dirname(__file__) + '/wfs_' +  request.lower() + '_expected.txt', 'w+')
+        f.write(expected)
+        f.close()
+        f = open(os.path.dirname(__file__) + '/wfs_' +  request.lower() + '_response.txt', 'w+')
+        f.write(response)
+        f.close()
+        """
+        response = re.sub(RE_STRIP_PATH, '', response)
+        expected = re.sub(RE_STRIP_PATH, '', expected)
+        self.assertEqual(response, expected, msg="request %s failed.\n Query: %s\n Expected:\n%s\n\n Response:\n%s" % (query_string, request, expected, response))
+
+    def test_project_wfs(self):
+        """Test some WMS request"""
+        for request in ('GetCapabilities', 'DescribeFeatureType'):
+            self.wfs_request_compare(request)
+
+    def wfs_getfeature_compare(self, requestid, request):
+        project = self.testdata_path + "test+project_wfs.qgs"
+        assert os.path.exists(project), "Project file not found: " + project
+
+        query_string = 'MAP=%s&SERVICE=WFS&VERSION=1.0.0&REQUEST=%s' % (urllib.quote(project), request)
+        header, body = [str(_v) for _v in self.server.handleRequest(query_string)]
+        self.assert_headers(header, body)
+        response = header + body
+        f = open(self.testdata_path + 'wfs_getfeature_'+ requestid + '.txt')
+        expected = f.read()
+        f.close()
+        # Store the output for debug or to regenerate the reference documents:
+        """
+        f = open(os.path.dirname(__file__) + '/wfs_getfeature_' +  requestid + '_expected.txt', 'w+')
+        f.write(expected)
+        f.close()
+        f = open(os.path.dirname(__file__) + '/wfs_getfeature_' +  requestid + '_response.txt', 'w+')
+        f.write(response)
+        f.close()
+        """
+        response = re.sub(RE_STRIP_PATH, '', response)
+        expected = re.sub(RE_STRIP_PATH, '', expected)
+        self.assertEqual(response, expected, msg=u"request %s failed.\n Query: %s\n Expected:\n%s\n\n Response:\n%s"
+                                                 % (query_string,
+                                                    request,
+                                                    unicode(expected, errors='replace'),
+                                                    unicode(response, errors='replace')))
+
+    def test_getfeature(self):
+        tests = []
+        tests.append(('nobbox', u'GetFeature&TYPENAME=testlayer&SRSNAME=EPSG:4326'))
```

Comment:
QGIS Server does not support SRSNAME parameter. SRSNAME is  a WFS 1.1.0 GetFEature Request parameter. QGIS Server does not implement WFS 1.1.0.
Can you update the code ?